modal/parallel_map.py

```diff
@@ -249,9 +250,7 @@ async def poll_outputs():
 
         assert len(received_outputs) == 0
 
-    response_gen = stream.merge(drain_input_generator(), pump_inputs(), poll_outputs())
-
-    async with response_gen.stream() as streamer:
+    async with aclosing(async_merge(drain_input_generator(), pump_inputs(), poll_outputs())) as streamer:
```

Comment:
Same here, the inputs to async_merge would all benefit from `aclosing` to make sure we close those generators in a timely fashion